includes/collection/class-outbox.php

```diff
@@ -38,17 +42,32 @@ public static function add( $activity_object, $activity_type, $user_id, $content
 				break;
 		}
 
+		$title                 = $activity_object->get_name() ?? $activity_object->get_content();
+		$activitypub_object_id = $activity_object->get_id();
+
+		if ( ! $title && is_activity( $activity_object ) && $activity_object->get_object() instanceof \Activitypub\Activity\Base_Object ) {
+			$title                 = $activity_object->get_object()->get_name() ?? $activity_object->get_object()->get_content();
+			$activitypub_object_id = $activity_object->get_object()->get_id();
+		}
+
 		$outbox_item = array(
 			'post_type'    => self::POST_TYPE,
-			'post_title'   => $activity_object->get_id(),
+			'post_title'   => sprintf(
+				/* translators: 1. Activity type, 2. Object type, 3. Object Title or Excerpt */
+				__( '[%1$s] %2$s: %3$s', 'activitypub' ),
+				$activity_type,
+				$activity_object->get_type(),
+				\wp_trim_words( $title, 5 )
+			),
 			'post_content' => wp_slash( $activity_object->to_json() ),
 			// ensure that user ID is not below 0.
 			'post_author'  => \max( $user_id, 0 ),
 			'post_status'  => 'pending',
 			'meta_input'   => array(
-				'_activitypub_activity_type'     => $activity_type,
-				'_activitypub_activity_actor'    => $actor_type,
-				'activitypub_content_visibility' => $content_visibility,
+				'_activitypub_object_id'         => \esc_url_raw( $activitypub_object_id ),
+				'_activitypub_activity_type'     => \esc_attr( $activity_type ),
+				'_activitypub_activity_actor'    => \esc_attr( $actor_type ),
+				'activitypub_content_visibility' => \esc_attr( $content_visibility ),
```

Comment:
What makes it necessary to escape these three?

Reply:
shouldn't we escape anything to go into the db?

Reply:
Happy to remove it

Reply:
Ah, we already do that in the register_post_meta call! 🫣

Reply:
Yeah, the naming of esc_url_raw() compared with the rest of the `esc_*()` family of functions is confusing. [This VIP doc](https://docs.wpvip.com/security/validating-sanitizing-and-escaping/) is a good primer on the difference between sanitization and escaping in WP and links to [a handy list of available sanitization functions](https://developer.wordpress.org/apis/security/sanitizing/#sanitization-functions).

`esc_url_raw()` calls `sanitize_url()` which then calls `esc_url()` with the `db` context.

Reply:
Let's also remove them from `invalidate_existing_items()`

Reply:
done